owncloudData/src/main/java/com/owncloud/android/data/files/repository/OCFileRepository.kt

```diff
@@ -579,24 +583,49 @@ class OCFileRepository(
         localFileDataSource.cleanWorkersUuid(fileId)
     }
 
-    private fun deleteLocalFolderRecursively(ocFile: OCFile, onlyFromLocalStorage: Boolean) {
+    private fun deleteLocalFolderRecursively(ocFile: OCFile, onlyFromLocalStorage: Boolean, keepAvailableOfflineFile: Boolean = false) {
         val folderContent = localFileDataSource.getFolderContent(ocFile.id!!)
 
         // 1. Remove folder content recursively
         folderContent.forEach { file ->
-            if (file.isFolder) {
-                deleteLocalFolderRecursively(ocFile = file, onlyFromLocalStorage = onlyFromLocalStorage)
+            if (keepAvailableOfflineFile) {
+                if (!file.isAvailableOffline) {
+                    deleteLocalFolderRecursivelyOrLocalFile(
+                        file = file,
+                        onlyFromLocalStorage = onlyFromLocalStorage,
+                    )
+                }
             } else {
-                deleteLocalFile(ocFile = file, onlyFromLocalStorage = onlyFromLocalStorage)
+                deleteLocalFolderRecursivelyOrLocalFile(
+                    file = file,
+                    onlyFromLocalStorage = onlyFromLocalStorage,
+                )
             }
```

Comment:
The content of the `if` and the `else` is the same. That doesn't make sense, take a look at the algorithm